breathecode/utils/decorators/has_permission.py

```diff
@@ -180,7 +196,47 @@ def build_context(**opts):
                     raise e
 
                 if format == 'html':
-                    return render_message(request, str(e), status=402)
+                    from breathecode.payments.models import Subscription, PlanOffer, PlanFinancing
+
+                    service = None
+                    if 'service_slug' in kwargs:
+                        service = kwargs['service_slug']
+
+                    renovate_consumables = {}
+
+                    subscription = Subscription.objects.filter(
+                        user=request.user,
+                        plans__mentorship_service_set__mentorship_services__slug=service).first()
+
```

Comment:
It only works with mentorship services and it should fail if the `consumer` has custom rules of comsuption, causing a message error related to consumables within a resource that was marked as free for users with that permission